tests/zsa.rs

```diff
@@ -7,20 +7,24 @@ use orchard::bundle::Authorized;
 use orchard::issuance::{verify_issue_bundle, IssueBundle, IssueInfo, Signed, Unauthorized};
 use orchard::keys::{IssuanceAuthorizingKey, IssuanceValidatingKey};
 use orchard::note::{AssetBase, ExtractedNoteCommitment};
-use orchard::note_encryption_v3::OrchardDomainV3;
+
 use orchard::tree::{MerkleHashOrchard, MerklePath};
 use orchard::{
     builder::Builder,
     bundle::Flags,
     circuit::{ProvingKey, VerifyingKey},
     keys::{FullViewingKey, PreparedIncomingViewingKey, Scope, SpendAuthorizingKey, SpendingKey},
+    note_encryption::OrchardDomainBase,
+    orchard_flavor::OrchardZSA,
     value::NoteValue,
     Address, Anchor, Bundle, Note,
 };
 use rand::rngs::OsRng;
 use std::collections::HashSet;
 use zcash_note_encryption_zsa::try_note_decryption;
 
+type OrchardDomainZSA = OrchardDomainBase<OrchardZSA>;
```

Comment:
From @PaulLaux:
EncDomainZsa

Reply:
In our discussion, we decided to keep the name `OrchardDomain` for now (along with `OrchardDomainBase`, `OrchardDomainVanilla`, and `OrchardDomainZSA`) because we have three traits - `OrchardDomain`, `OrchardCircuit`, and `OrchardHash`. This naming scheme with `Orchard` prefix looks consistent.